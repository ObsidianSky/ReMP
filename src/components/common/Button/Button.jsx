import React from 'react';
import PropTypes from 'prop-types';
import './Button.scss'

const Button = ({ className, onClick, children, text}) => (
    <button className={`btn ${className}`} onClick={onClick} type="button">
        {children || text}
    </button>
);

Button.defaultProps = {
    text: ''
};

Button.propTypes = {
    text: PropTypes.string,
    onClick: PropTypes.func.isRequired,
    className: PropTypes.string,
    children: PropTypes.oneOfType([
        PropTypes.arrayOf(PropTypes.node),
        PropTypes.node,
    ])
};

export default Button;