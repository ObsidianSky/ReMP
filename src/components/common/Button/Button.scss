@import '../../../assets/styles/variables';

.btn {
  display: inline-block;
  border: none;
  background: #fff;
  color: $pink;
  outline: none;
  font-family: 'Open Sans', sans-serif;
  font-size: 15px;
  text-transform: uppercase;
  font-weight: 600;
  padding: 7px 16px;

  &:hover {
    color: #fff;
    background: $pink;
    cursor: pointer;
  }

  &--link {
    padding: 0;
    background: none;
    color: $text-black;
    text-transform: none;

    &:hover, &.active {
      color: $pink;
      background: none;
    }
  }

  &--narrow {
    font-size: 14px;
    padding: 4px 7px;
  }

  &--wide {
    padding: 7px 40px;
  }

  &--gray {
    background: $gray;
    color: #fff;

    &:hover, &.active {
      background: $pink;
    }
  }

  &--pink {
    color: #fff;
    background: $pink;

    &:hover {
      background: #fff;
      color: $pink;
    }
  }
}



