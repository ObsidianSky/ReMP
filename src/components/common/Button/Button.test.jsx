import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Button from './Button';

describe('Button', () => {
    let props;
    let onClick;

    beforeEach(() => {
        onClick = jest.fn();

        props = {
            text: 'text',
            className: 'className',
            onClick
        }
    });

    describe('when we do NOT have children', () => {
        test('should renders component with text', () => {
            const wrapper = shallow(<Button { ...props }/>);

            expect(toJson(wrapper)).toMatchSnapshot();
        });
    });

    describe('when we have children', () => {
        beforeEach(() => {
            props.children = <div>children</div>;
        });

        test('should renders component with children instead of text', () => {
            const wrapper = shallow(<Button { ...props }/>);

            expect(toJson(wrapper)).toMatchSnapshot();
        });
    });

    describe('when we click on button', () => {
        test('should call onclick callbeck', () => {
            const wrapper = shallow(<Button { ...props }/>);

            wrapper.find('button').simulate('click');

            expect(onClick).toHaveBeenCalled();
        });
    });
});
