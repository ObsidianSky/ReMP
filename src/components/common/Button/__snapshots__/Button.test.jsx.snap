// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button when we do NOT have children should renders component with text 1`] = `
<button
  className="btn className"
  onClick={[Function]}
  type="button"
>
  text
</button>
`;

exports[`Button when we have children should renders component with children instead of text 1`] = `
<button
  className="btn className"
  onClick={[Function]}
  type="button"
>
  <div>
    children
  </div>
</button>
`;
